pkg/workload/schemachange/error_screening.go

```diff
@@ -1694,3 +1616,73 @@ func (og *operationGenerator) databaseHasTablesWithPartitioning(
 			database),
 	)
 }
+
+// getUniqueConstraintsForTable returns the set of expressions associated with unique indexes
+// in the specified tableName.
+func getUniqueConstraintsForTable(
+	ctx context.Context, tx pgx.Tx, tableName string,
+) (constraints []string, err error) {
+	q := fmt.Sprint(`
+WITH tab_json AS (
+                    SELECT crdb_internal.pb_to_json(
+                            'desc',
+                            descriptor
+                           )->'table' AS t
+                      FROM system.descriptor
+                     WHERE id = $1::REGCLASS
+                  ),
+         columns_json AS (
+                        SELECT json_array_elements(t->'columns') AS c FROM tab_json
+                      ),
+         columns AS (
+                    SELECT (c->>'id')::INT8 AS col_id,
+                           IF(
+                            (c->'inaccessible')::BOOL,
+                            c->>'computeExpr',
+                            quote_ident(c->>'name')
+                           ) AS expr
+                      FROM columns_json
+                 ),
+         indexes_json AS (
+                         SELECT json_array_elements(t->'indexes') AS idx
+                           FROM tab_json
+                         UNION ALL SELECT t->'primaryIndex' FROM tab_json
+                      ),
+         unique_indexes AS (
+                            SELECT idx->'name' AS name,
+                                   json_array_elements(
+                                    idx->'keyColumnIds'
+                                   )::STRING::INT8 AS col_id
+                              FROM indexes_json
+															WHERE (idx->'unique')::BOOL
+                        ),
+         index_exprs AS (
+                        SELECT name, expr
+                          FROM unique_indexes AS idx
+                               INNER JOIN columns AS c ON idx.col_id = c.col_id
+                     )
+      SELECT '(' || array_to_string(array_agg(expr), ', ') || ')' AS final_expr                    
```

Comment:
This specific change (i.e. we don't use `SELECT ARRAY['(' || array_to_string(array_agg(expr), ', ') || ')'] AS final_expr`) had to happen because casting to an array implicitly improperly escaped the column names - leading us to use queries like `SELECT \"col blah2\" ...` instead of `SELECT "col blah2" ...`